### PR TITLE
3.17 version OHOS adaptation

### PR DESCRIPTION
**Pull request by IceBBBBB**

### CODE REVIEW

File: cmake/Modules/CocosCompileOptions.cmake [hunk 1]
```diff
@@ -5,11 +5,14 @@
...
  #MACOSX    =  MacOS X
  #LINUX      =   Linux
+ #OHOS = Ohos
  if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
      set(WINDOWS TRUE)
      set(SYSTEM_STRING "Windows Desktop")
  elseif(${CMAKE_SYSTEM_NAME} MATCHES "Android")
      set(SYSTEM_STRING "Android")
+ elseif(${CMAKE_SYSTEM_NAME} MATCHES "OHOS")
+     set(SYSTEM_STRING "OHOS")
```
Comment (minggo) [inline cmake/Modules/CocosCompileOptions.cmake]: Should change to `HarmonyOS Next`?

File: cocos/renderer/CCGLProgramState.cpp [hunk 2]
```diff
@@ -419,8 +419,11 @@ GLProgramState::GLProgramState()
 , _vertexAttribsFlags(0)
 , _glprogram(nullptr)
 , _nodeBinding(nullptr)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+, _backToForegroundlistener(nullptr)
+#endif
```
Comment (minggo) [inline cocos/renderer/CCGLProgramState.cpp]: Why only initialize it on OHOS?

File: cocos/renderer/CCMeshCommand.cpp [hunk 3]
```diff
@@ -55,10 +55,13 @@ MeshCommand::MeshCommand()
 , _glProgramState(nullptr)
 , _stateBlock(nullptr)
 , _textureID(0)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+, _rendererRecreatedListener(nullptr)
+#endif
```
Comment (minggo) [inline cocos/renderer/CCMeshCommand.cpp]: Why only initialize on OHOS?

File: cocos/scripting/lua-bindings/script/framework/device.lua [hunk 4]
```diff
@@ -58,6 +58,8 @@ elseif target == cc.PLATFORM_OS_WINRT then
     device.platform = "winrt"
 elseif target == cc.PLATFORM_OS_WP8 then
     device.platform = "wp8"
+elseif target == cc.PLATFORM_OS_OHOS then
+    device.platform = "harmony"
```
Comment (minggo) [inline cocos/scripting/lua-bindings/script/framework/device.lua]: Should change to `HarmonyOS Next`?

File: tests/lua-tests/src/controller.lua [hunk 5]
```diff
@@ -1,4 +1,7 @@
 
+-- jit off
+local jit = require("jit")
+jit.off()
```
Comment (minggo) [inline tests/lua-tests/src/controller.lua]: Why have to disable jit?

File: cocos/renderer/CCGLProgramState.cpp (hunk 2 above)
Comment (IceBBBBB) [reply cocos/renderer/CCGLProgramState.cpp]: modified

File: cocos/renderer/CCMeshCommand.cpp (hunk 3 above)
Comment (IceBBBBB) [reply cocos/renderer/CCMeshCommand.cpp]: modified

File: cmake/Modules/CocosCompileOptions.cmake (hunk 1 above)
Comment (IceBBBBB) [reply cmake/Modules/CocosCompileOptions.cmake]: modified

File: cocos/scripting/lua-bindings/script/framework/device.lua (hunk 4 above)
Comment (IceBBBBB) [reply cocos/scripting/lua-bindings/script/framework/device.lua]: modified

Comment (minggo) [reply cocos/scripting/lua-bindings/script/framework/device.lua]: Is it modified?

File: cocos/renderer/CCMeshCommand.cpp (hunk 3 above)
Comment (minggo) [reply cocos/renderer/CCMeshCommand.cpp]: Is it modified?

File: tests/lua-tests/src/controller.lua (hunk 5 above)
Comment (IceBBBBB) [reply tests/lua-tests/src/controller.lua]: modified

**Pull request merged**